### PR TITLE
Add a contributing section to the sway book

### LINKED ISSUE #940: Add a CONTRIBUTING.md
Just a reminder to add a `CONTRIBUTING.md` as we're currently missing one for this repo.

:thought_balloon: I wonder if we should have a **Contributing** chapter toward the end of the Sway book and point to that?

### PR DESCRIPTION
**Pull request by nfurfaro**
It's all in the title. I pulled inspiration from the fuel-core Contributing.md file.

close #940

### CODE REVIEW

File: docs/src/reference/contributing_to_sway.md [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+If you run into any difficulties getting started, you can always ask questions on our [Discord](https://discord.gg/xfpK4Pe).
+
+## Building and setting up a development workspace
+
+See the Sway book's [Introduction](../introduction/index.md) for instructions on installing and setting up the Sway toolchain..
+
+## Getting the repository
+
+```
+git clone https://github.com/FuelLabs/sway
```
Comment (nfurfaro) [inline docs/src/reference/contributing_to_sway.md:14]: Should we be directing contributors to fork the sway repo first?

File: Contributing.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+# contributing to Sway
+
+We welcome contributions to Sway!
+
+Please see the [Contributing To Sway](https://fuellabs.github.io/sway/latest/reference/contributing_to_sway.html) section of the Sway book for guidelines and instructions to help you get started.
```
Comment (adlerjohn) [inline Contributing.md:5]: Is there really a need for a whole new file? Could this be at the end of the readme?

File: docs/src/reference/contributing_to_sway.md [hunk 3]
```diff
@@ -0,0 +1,99 @@
+# Contributing To Sway
+
+Thanks for your interest in contributing to Sway! This document outlines the process for installing and setting up the Sway toolchain, as well as some conventions on contributing to Sway.
```
Comment (adlerjohn) [inline docs/src/reference/contributing_to_sway.md:3]:
```suggestion
Thanks for your interest in contributing to Sway! This document outlines the process for installing and setting up the Sway toolchain for development, as well as some conventions on contributing to Sway.
```

File: docs/src/reference/contributing_to_sway.md [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+
+If you run into any difficulties getting started, you can always ask questions on our [Discord](https://discord.gg/xfpK4Pe).
+
+## Building and setting up a development workspace
+
+See the Sway book's [Introduction](../introduction/index.md) for instructions on installing and setting up the Sway toolchain..
```
Comment (adlerjohn) [inline docs/src/reference/contributing_to_sway.md:9]: You're already in The Sway Book, so no need for saying "see the Sway book's..."

File: docs/src/reference/contributing_to_sway.md [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+
+## Getting the repository
+
+```
```
Comment (adlerjohn) [inline docs/src/reference/contributing_to_sway.md:13]: And elsewhere
````suggestion
```sh
````

File: docs/src/reference/contributing_to_sway.md [hunk 6]
```diff
@@ -0,0 +1,99 @@
...
+git clone https://github.com/FuelLabs/sway
+cd sway
+```
+
+## Building and testing
+
+The following steps will run the sway test suite and ensure that everything is set up correctly.
+
+First, open a new terminal and start fuel-core with:
```
Comment (adlerjohn) [inline docs/src/reference/contributing_to_sway.md:22]:
```suggestion
First, open a new terminal and start `fuel-core` with:
```

File: docs/src/reference/contributing_to_sway.md [hunk 7]
```diff
@@ -0,0 +1,99 @@
...
+
+```
+fuel-core
+```
+
+Then, open a second terminal and run:
+
+```
+cd sway/test
+cargo run
```
Comment (adlerjohn) [inline docs/src/reference/contributing_to_sway.md:32]: Shorter would be
```suggestion
cargo run --bin test
```

File: docs/src/reference/contributing_to_sway.md [hunk 8]
```diff
@@ -0,0 +1,99 @@
...
+```
+
+After the test suite runs, you should see:
+
+> ---
+>
+> Tests passed.\
+> _n_ tests run (0 skipped)
```
Comment (adlerjohn) [inline docs/src/reference/contributing_to_sway.md:40]:
````suggestion
```console
Tests passed.
_n_ tests run (0 skipped)
```
````

File: docs/src/reference/contributing_to_sway.md [hunk 9]
```diff
@@ -0,0 +1,99 @@
...
+Congratulations! You've now got everything setup and are ready to start making contributions.
+
+## Finding something to work on
+
+There are many ways in which you may contribute to the Sway project, some of which involve coding knowledge and some which do not. A few examples include:
+
+- Reporting bugs
+- Adding documentation to the Sway book
+- Adding new features or bugfixes for which there is already an open issue
+- making Feature requests
```
Comment (adlerjohn) [inline docs/src/reference/contributing_to_sway.md:51]:
```suggestion
- Making feature requests
```

File: docs/src/reference/contributing_to_sway.md [hunk 10]
```diff
@@ -0,0 +1,99 @@
...
+
+Check out our [Help Wanted](https://github.com/FuelLabs/sway/issues?q=is%3Aopen+is%3Aissue+label%3A%22help+wanted%22), [Sway Book](https://github.com/FuelLabs/sway/issues?q=is%3Aopen+is%3Aissue+label%3A%22The+Sway+Book%22) or [Good First Issue](https://github.com/FuelLabs/sway/issues?q=is%3Aopen+is%3Aissue+label%3A%22good+first+issue%22) issues to find a suitable task.
+
+If you are planning something big, for example, related to multiple components or changes current behaviors, make sure to open an issue to discuss with us before going on.
```
Comment (adlerjohn) [inline docs/src/reference/contributing_to_sway.md:55]:
```suggestion
If you are planning something big, for example, related to multiple components or changes current behaviors, make sure to open an issue to discuss with us before starting on the implementation.
```

File: docs/src/reference/contributing_to_sway.md [hunk 11]
```diff
@@ -0,0 +1,99 @@
...
+  - If there are conflicts, you can merge and resolve them locally. Then push to your PR branch.
+    Any changes to the branch will require a re-review.
+- Our CI system (Github Actions) automatically tests all authorized pull requests.
+- Use Github to merge the PR once approved.
+
+Thanks for your contributions!
+
+### Linking issues
+
+Pull Requests should be linked to at least one issue in the same repo.
```
Comment (adlerjohn) [inline docs/src/reference/contributing_to_sway.md:81]:
```suggestion
Pull requests should be linked to at least one issue in the same repo.
```

File: docs/src/reference/contributing_to_sway.md (hunk 1 above)
Comment (adlerjohn) [reply docs/src/reference/contributing_to_sway.md:14]: Probably, yes

File: README.md [hunk 12]
```diff
@@ -44,3 +44,9 @@ Confirm the Sway toolchain built successfully:
 ```sh
 cargo run --bin forc -- --help
 ```
+
+## contributing to Sway
```
Comment (adlerjohn) [inline README.md:48]:
```suggestion
## Contributing to Sway
```

File: docs/src/reference/contributing_to_sway.md [hunk 13]
```diff
@@ -0,0 +1,101 @@
+# Contributing To Sway
+
+Thanks for your interest in contributing to Sway! This document outlines the process for installing and setting up the Sway toolchain for development, as well as some conventions on contributing to Sway.
+
+If you run into any difficulties getting started, you can always ask questions on our [Discord](https://discord.gg/xfpK4Pe).
+
+## Building and setting up a development workspace
+
+See the [introduction](../introduction/index.md) section for instructions on installing and setting up the Sway toolchain..
```
Comment (mitchmindtree) [inline docs/src/reference/contributing_to_sway.md:9]:
```suggestion
See the [introduction](../introduction/index.md) section for instructions on installing and setting up the Sway toolchain.
```

File: docs/src/reference/contributing_to_sway.md [hunk 14]
```diff
@@ -0,0 +1,101 @@
...
+```sh
+git clone https://github.com/FuelLabs/sway
+cd sway
+```
+
+## Building and testing
+
+The following steps will run the sway test suite and ensure that everything is set up correctly.
+
+First, open a new terminal and start `fuel-core` with:
```
Comment (mitchmindtree) [inline docs/src/reference/contributing_to_sway.md:25]: Maybe prior to this we should have a section on installing `fuel-core` along with a brief sentence explaining why it's required to run the tests? Or perhaps provide a link if we already have a guide on `fuel-core` setup elsewhere?

File: docs/src/reference/contributing_to_sway.md [hunk 15]
```diff
@@ -0,0 +1,101 @@
...
+## Contribution flow
+
+This is a rough outline of what a contributor's workflow looks like:
+
+- Make sure what you want to contribute is already tracked as an issue.
+  - We may discuss the problem and solution in the issue.
+- Create a Git branch from where you want to base your work. This is usually master.
+- Write code, add test cases, and commit your work.
+- Run tests and make sure all tests pass.
+- If the PR contains any breaking changes, add the breaking label to your PR.
```
Comment (mitchmindtree) [inline docs/src/reference/contributing_to_sway.md:68]: Let's remember to add a "breaking" label for the sway repo after landing this.

**Review by mitchmindtree — CHANGES_REQUESTED** [review]
This reads really well! Just a few minor nits and I think this will be good to go.

File: docs/src/reference/contributing_to_sway.md (hunk 14 above)
Comment (adlerjohn) [reply docs/src/reference/contributing_to_sway.md:25]: The Sway Book should already talk about installing `fuel-core`, so a link should be sufficient.

Comment (nfurfaro) [reply docs/src/reference/contributing_to_sway.md:25]: The section above, `Building and setting up a development workspace` already has a link to the sway book's [introduction](https://fuellabs.github.io/sway/latest/introduction/index.html) which _should_ be sufficient?

**Review by mitchmindtree — APPROVED** [review]

**Review by adlerjohn — APPROVED** [review]

**Pull request merged**